core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -635,8 +639,10 @@ private void seekInternalWithCachingPartiallyReadBlock(long pos) throws IOExcept
     } else {
       mPos = pos / mBlockSize * mBlockSize;
       updateStreams();
-      if (mCurrentCacheStream != null) {
-        cacheCurrentBlockToPos(pos);
+      if (mCurrentBlockInStream != null
+          && mCurrentBlockInStream.Source() != BlockInStreamSource.LOCAL) {
```

Comment:
Why is this condition different from before? `mCurrentBlockInStream` vs `mCurrentCacheStream`

Reply:
Because `mCurrentCacheStream  != null` is insufficient to denote if partial caching is needed. i.e. when reading from UFS, the cache stream is null, but we still need partial caching.

Reply:
can you javadoc this . it is actually quite tricky

Reply:
Rewrote the javadoc